packages/graph/hash_graph/lib/graph/src/shared/identifier/mod.rs

```diff
@@ -13,14 +23,177 @@ pub mod account;
 pub mod knowledge;
 pub mod ontology;
 
-pub type Timestamp = DateTime<Utc>;
+#[derive(
+    Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, ToSql,
+)]
+#[serde(transparent)]
+#[postgres(transparent)]
+pub struct DecisionTimestamp(DateTime<Utc>);
+
+impl fmt::Display for DecisionTimestamp {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> std::fmt::Result {
+        fmt::Display::fmt(&self.0, fmt)
+    }
+}
+
+impl FromStr for DecisionTimestamp {
+    type Err = chrono::ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(DateTime::from_str(s)?))
+    }
+}
+
+#[derive(
+    Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize, ToSql,
+)]
+#[serde(transparent)]
+#[postgres(transparent)]
+pub struct TransactionTimestamp(DateTime<Utc>);
+
+impl fmt::Display for TransactionTimestamp {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> std::fmt::Result {
+        fmt::Display::fmt(&self.0, fmt)
+    }
+}
+
+impl FromStr for TransactionTimestamp {
+    type Err = chrono::ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(Self(DateTime::from_str(s)?))
+    }
+}
 
 #[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
-pub struct Timespan {
-    pub from: Timestamp,
-    pub to: Option<Timestamp>,
+pub(crate) struct Timespan {
+    start: Bound<DateTime<Utc>>,
+    end: Bound<DateTime<Utc>>,
+}
+
+impl FromSql<'_> for Timespan {
+    fn from_sql(_: &Type, buf: &[u8]) -> Result<Self, Box<dyn std::error::Error + Sync + Send>> {
+        fn parse_bound(
+            bound: &RangeBound<Option<&[u8]>>,
+        ) -> Result<Bound<DateTime<Utc>>, Box<dyn std::error::Error + Send + Sync>> {
+            Ok(match bound {
+                RangeBound::Inclusive(None) | RangeBound::Exclusive(None) => {
+                    unimplemented!("null ranges are not supported")
+                }
+                RangeBound::Inclusive(Some(bytes)) => {
+                    let timestamp = DateTime::from_sql(&Type::TIMESTAMPTZ, bytes)?;
+                    Bound::Included(timestamp)
+                }
+                RangeBound::Exclusive(Some(bytes)) => {
+                    let timestamp = DateTime::from_sql(&Type::TIMESTAMPTZ, bytes)?;
+                    Bound::Excluded(timestamp)
+                }
+                RangeBound::Unbounded => Bound::Unbounded,
+            })
+        }
+
+        fn is_negative_infinity(bound: &RangeBound<Option<&[u8]>>) -> io::Result<bool> {
+            Ok(match bound {
+                RangeBound::Inclusive(Some(mut bytes)) | RangeBound::Exclusive(Some(mut bytes)) => {
+                    bytes.read_i64::<BigEndian>()? == i64::MIN
```

Comment:
Are we sure about the endianness, is it defined in the Postgres spec/code anywhere?

Reply:
the endianness is used in the `postgres-protocol` crate. However, I was able to find a function, which does that logic for us, so I was able to remove the `byteorder` crate.